### DISCUSSION TITLE
Enable to access function on custom field

### DISCUSSION THREAD

**Discussion by hexionaut** [filamentphp/filament#8861] [Help] [answered]
Hi,

I created a custom field using a blade template and a livewire class (made by extending `Field`).

Everything goes ok when I read data from that class in my blade component. However, when i want to make a button interactive by doing this : 

BLADE
```
<x-filament::button color="danger" wire:click="banAllAccountsWithIdenticalRegistrationIpAndBlockIp">
    Ban all users
</x-filament::button>
``` 

CLASS
```
public function banAllAccountsWithIdenticalRegistrationIpAndBlockIp()
{
    Account::where('registration_ip', $this->getRegistrationIp())->update([
        'is_banned => true'
    ]);

    BlockedIp::create([
        'ip' => $this->getRegistrationIp(),
    ]);
}
```

I am having this error : 

```
Unable to call component method. Public method [banAllAccountsWithIdenticalRegistrationIpAndBlockIp] not found on component: [domain.admin.resources.account-resource.pages.edit-account]
```

Any idea on how to fix this ? Don't know why it ask me about edit-account as this method is in the custom field class.

Thanks for helping !

**Comment by sprtk-ches** [comment]
There's a similar question [here](https://github.com/filamentphp/filament/discussions/8159)

Unless something has changed, the suggested method is to use [Actions ](https://filamentphp.com/docs/3.x/forms/actions#adding-an-action-to-a-custom-form-component)

**Reply by hexionaut** [reply]
Thanks for you answer however I am on v2 of Filament. Do you have any idea then ?

**Comment by hexionaut** [comment, marked as answer]
Ok. So I found a solution.

Added the `banAllAccountsWithIdenticalRegistrationIpAndBlockIp` function into my `EditAccount` then used it in my component.
